src/lib/components/events/event-item.svelte

```diff
@@ -1,135 +1,208 @@
 <script lang="ts">
   import { onMount } from 'svelte';
   import type { AcmEvent } from '$lib/ical/parse';
-  import AcmButton from '$lib/components/utils/acm-button.svelte';
 
   export let info: AcmEvent;
 
-  let isActive = false;
   let isRecurring: boolean = info.recurring;
-  let isSuccessfullyCopied = false;
-  let anchor: HTMLDivElement;
-
-  function copyEventLink(slug: string) {
-    const url = [location.origin, location.pathname, '#', slug].join('');
-    // Copying text to the clipboard: https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText
-    navigator.clipboard.writeText(url).then(() => {
-      isSuccessfullyCopied = true;
-      setTimeout(() => (isSuccessfullyCopied = false), 2e3);
-    });
-  }
+  let anchor: HTMLElement;
+  let details: HTMLDetailsElement;
 
   onMount(() => {
-    isActive = location.hash === `#${info.slug}`;
-    if (isActive) {
+    if (location.hash === `#${info.slug}`) {
       anchor.scrollIntoView({
         behavior: 'smooth',
         block: 'start',
         inline: 'center',
       });
+      // This has the same styling as :target, and :target doesn't want to work
+      // for dynamically-added elements.
+      details.open = true;
     }
   });
 </script>
 
 <div class="event-box">
-  <div class="anchor" id={info.slug} class:active={isActive} bind:this={anchor} />
-  <div class="event-card">
-    <p class="event-date">
-      <span class="mid">
-        {info.month}
-        {info.day}
-      </span>
-      {#if isRecurring}
-        <p class="event-recurring">RECURRING</p>
-      {/if}
-    </p>
-    <h3
-      class="headers"
-      class:copied={isSuccessfullyCopied}
-      on:click={() => copyEventLink(info.slug)}>
-      {info.summary}
-    </h3>
-    <div>
-      <p class="event-time">{info.time} PT</p>
-      <p class="event-location">
-        {info.location === 'Discord' || info.location === 'Zoom'
-          ? `Hosted on ${info.location}`
-          : info.location}
+  <!-- Workaround for the top panel covering the event card's anchor. -->
+  <div class="anchor" id={info.slug} bind:this={anchor} />
+  <details class="event-card" bind:this={details}>
+    <summary class="event-body">
+      <div class="event-name">
+        <h2 class="headers">
+          <a href="#{info.slug}">
+            {info.summary}
+          </a>
+        </h2>
```

Comment:
The inner link can be left out from this `<h2>` element since it interferes with the opening of the `<details>` element when the visitor goes to click the title.

Reply:
There should be another way to obtain the anchor link if we want to remove the `<a>` tag. Perhaps a copy button inside.

Reply:
Good call! Here is a new bug for that: #281.

Reply:
Removed the <a> tag in <https://github.com/EthanThatOneKid/acmcsuf.com/pull/280/commits/219dbbd436a8ca7433a1313a003ff2795d7f2879>.